### PR TITLE
Add NCEPLIBS-bufr

### PR DESCRIPTION
**Pull request by darothen**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by darothen** [comment]
@jbathegit I'm developing a tool which would greatly benefit from [NOAA-EMC/NCEPLIBS-bufr](https://github.com/NOAA-EMC/NCEPLIBS-bufr) being available on `conda-forge`, so I'm taking the liberty of creating a feedstock it. Since you're listed as the code manager on that repo, I thought it prudent that you were listed as a maintainer here.

Note that I need to work through tweaking the build to create a dynamic library before completing this (e.g. https://github.com/conda-forge/cfep/blob/main/cfep-18.md). So this isn't immediately ready to ping for review, but it would be great if you could comment here confirming that you can help as a maintainer.

**Comment by darothen** [comment]
The current osx64 build fails quite reproducibly for a very specific reason, and I was able to debug it locally on my MacBook.

Take the test **intest7_4** as an example. In this test, `subroutine errwrt` is locally overridden within the actual test program, and `module Share_errstr_intest7` is used as a shim to allow the this subroutine to be called in the actual NCEPLIBS-bufr library routines. However, on platform osx64 something is going awry and the local override of the subroutine isn't being called, so we never populate the variable errstr (defined in `module Share_errstr_intest7`). Consequently, we fail the error check on L66 after we call `isetprm ( 'DUMMY', 20 )`; errstr is never set with the proper error message so the `index` function doesn't return 0, and we stop the program.

Not entirely sure how to fix this; it works totally fine on linux_64 (also tested locally, on a different machine). We _could_ just skip the handful of test cases that use this pattern and not even worry about the problem in the first place - all we care about is that the Python bindings work, and those tests pass just fine. I'm inclined to do that to avoid making any upstream changes to the NCEPLIBS-bufr code, unless @jbathegit has strong feelings either way.

**Comment by darothen** [comment]
Following the discussion at NOAA-EMC/NCEPLIBS-bufr#589, I'm making the following changes:

1. Removing @jbathegit as a proposed maintainer - @darothen will be the sole maintainer.
2. Modifying tests to suppress the handful that fail on osx64 due to the dynamic library patching issue as documented previously.

File: recipes/NCEPLIBS-bufr/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+source:
+  url: https://github.com/NOAA-EMC/NCEPLIBS-bufr/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 525f26238dba6511a453fc71cecc05f59e4800a603de2abbbbfb8cbb5adf5708
+
+build:
+  # NOTE: No Windows support in initial feedstock (known use cases are only osx and linux)
+  skip: True  # [win]
+  # NOTE: Only support recent Python releases. Cannot support Python==3.12 because build
+  # system in NCEPLIBS-bufr_v12.0.1 uses a hard-coded dependency on distutils, which was
```
Comment (ocefpaf) [inline recipes/NCEPLIBS-bufr/meta.yaml:16]: Fortran extensions are in a pickle with py312 :-/
They need to move to cmake or meson and that is not an easy change. Let me know if you figure out a path with least friction to solve this!

Comment (darothen) [reply recipes/NCEPLIBS-bufr/meta.yaml:16]: The good news is that this particular package does have an improvement to its cmake-based build already merged into trunk, so the next time they cut a release we'll be able to support Py312. But yeah, wish there was a generic way (maybe a back-port of distutils?)

**Pull request merged**